Some combinations of features fail to build

#### What version of regex are you using?

`1.5.2`, or with `regex-syntax` `0.6.24` with rust version `rustc 1.53.0-nightly (42816d61e 2021-04-24`

#### Describe the bug at a high level.

With default features disabled some feature combinations fail to compile.

#### What are the steps to reproduce the behavior?

For your `Cargo.toml`:

```toml
[package]
name="test"
version = "0.1.0"
edition = "2018"

[dependencies]
regex-syntax = { version = "0.6.24", default-features = false, features = ["unicode-perl"] }
```

This will also occur if you select the same feature set through `regex`:

```toml
[package]
name="test"
version = "0.1.0"
edition = "2018"

[dependencies]
regex = { version = "1.5.2", default-features = false, features = ["std", "unicode-perl"] }
```

#### What is the actual behavior?

```
  Compiling regex-syntax v0.6.24
error[E0433]: failed to resolve: use of undeclared crate or module `unicode_tables`
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:351:13
    |
351 |         use unicode_tables::perl_space::WHITE_SPACE;
    |             ^^^^^^^^^^^^^^ use of undeclared crate or module `unicode_tables`

error[E0433]: failed to resolve: use of undeclared crate or module `unicode_tables`
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:375:13
    |
375 |         use unicode_tables::perl_decimal::DECIMAL_NUMBER;
    |             ^^^^^^^^^^^^^^ use of undeclared crate or module `unicode_tables`

error[E0425]: cannot find value `WHITE_SPACE` in this scope
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:352:22
    |
352 |         Ok(hir_class(WHITE_SPACE))
    |                      ^^^^^^^^^^^ not found in this scope
    |
help: consider importing this constant
    |
1   | use crate::unicode_tables::perl_space::WHITE_SPACE;
    |

error[E0425]: cannot find value `DECIMAL_NUMBER` in this scope
   --> /home/jess/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.24/src/unicode.rs:376:22
    |
376 |         Ok(hir_class(DECIMAL_NUMBER))
    |                      ^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this constant
    |
1   | use crate::unicode_tables::perl_decimal::DECIMAL_NUMBER;
    |

error: aborting due to 4 previous errors

Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
error: could not compile `regex-syntax`
```

#### What is the expected behavior?

The program should compile.

--- a/regex-syntax/src/unicode.rs
+++ b/regex-syntax/src/unicode.rs
@@ -348,7 +348,7 @@ pub fn perl_space() -> Result<hir::ClassUnicode> {
 
     #[cfg(all(feature = "unicode-perl", not(feature = "unicode-bool")))]
     fn imp() -> Result<hir::ClassUnicode> {
-        use unicode_tables::perl_space::WHITE_SPACE;
+        use crate::unicode_tables::perl_space::WHITE_SPACE;
         Ok(hir_class(WHITE_SPACE))
     }
 
@@ -372,7 +372,7 @@ pub fn perl_digit() -> Result<hir::ClassUnicode> {
 
     #[cfg(all(feature = "unicode-perl", not(feature = "unicode-gencat")))]
     fn imp() -> Result<hir::ClassUnicode> {
-        use unicode_tables::perl_decimal::DECIMAL_NUMBER;
+        use crate::unicode_tables::perl_decimal::DECIMAL_NUMBER;
         Ok(hir_class(DECIMAL_NUMBER))
     }
 

--- a/regex-syntax/test
+++ b/regex-syntax/test
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # This is a convenience script for running a broad swath of the syntax tests.
 echo "===== DEFAULT FEATURES ==="
 cargo test

--- a/test
+++ b/test
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # This is a convenience script for running a broad swath of tests across
 # features. We don't test the complete space, since the complete space is quite
 # large. Hopefully once we migrate the test suite to better infrastructure

